src/domains/migration/hooks/use-meta-mask.ts

```diff
@@ -45,14 +25,17 @@ function isMetaMaskSupportedBrowser() {
 	return isCompatible && !isMobile;
 }
 
+const networkData = polygonNetworkData();
+
 export const useMetaMask = () => {
 	const [initialized, setInitialized] = useState<boolean>(false);
 	const [chainId, setChainId] = useState<number>();
 	const [account, setAccount] = useState<string | null>();
 	const [ethereumProvider, setEthereumProvider] = useState<ethers.providers.Web3Provider>();
 	const [connecting, setConnecting] = useState<boolean>(false);
 	const [switching, setSwitching] = useState<boolean>(false);
-	const isOnPolygonNetwork = useMemo(() => chainId === Number.parseInt(polygonNetworkData.chainId), [chainId]);
+
+	const isOnValidNetwork = useMemo(() => chainId === Number.parseInt(networkData.chainId), [chainId]);
```

Comment:
```suggestion
	const isOnValidNetwork = chainId === Number.parseInt(networkData.chainId);
```

since this is a simple comparison we can probably get rid of the memoization 🤔

Reply:
@dated the thing is that chain id is dynamic

Reply:
yep, setting the chainId causes a rerender, during which `isOnValidNetwork` is recomputed. what i meant is that this is such an inexpensive operation, that the memoization is unwarranted :oleglala: